src/Tools/dotnet-monitor/Egress/AzureBlobEgressFactory.cs

```diff
@@ -0,0 +1,117 @@
+            Dictionary<string, string> egressProperties,
+            out ConfiguredEgressProvider provider)
+        {
+            var options = providerSection.Get<ConfigurationOptions>();
+
+            if (string.IsNullOrEmpty(options.AccountKey) &&
+                !string.IsNullOrEmpty(options.AccountKeyName))
+            {
+                if (egressProperties.TryGetValue(options.AccountKeyName, out string key))
+                {
+                    options.AccountKey = key;
+                }
+            }
+
+            if (string.IsNullOrEmpty(options.SharedAccessSignature) &&
+                !string.IsNullOrEmpty(options.SharedAccessSignatureName))
+            {
+                if (egressProperties.TryGetValue(options.SharedAccessSignatureName, out string signature))
+                {
+                    options.SharedAccessSignature = signature;
+                }
+            }
+
+            if (!TryValidateOptions(options, providerName))
+            {
+                provider = null;
+                return false;
+            }
+
+            provider = new Provider(options, _loggerFactory);
+            return true;
+        }
+
+        private class Provider : ConfiguredEgressProvider
+        {
+            private readonly ILoggerFactory _loggerFactory;
+            private readonly AzureBlobEgressProviderOptions _options;
+
+            public Provider(AzureBlobEgressProviderOptions options, ILoggerFactory loggerFactory)
+            {
+                _options = options;
+                _loggerFactory = loggerFactory;
+            }
+
+            public override async Task<EgressResult> EgressAsync(
+                Func<CancellationToken, Task<Stream>> action,
+                string fileName,
+                string contentType,
+                IEndpointInfo source,
+                CancellationToken token)
+            {
+                // TODO: Add metadata based on source
+                var streamOptions = new AzureBlobEgressStreamOptions();
+                streamOptions.ContentType = contentType;
+
+                var provider = new AzureBlobEgressProvider(_options, _loggerFactory.CreateLogger<AzureBlobEgressProvider>());
+                string blobUri = await provider.EgressAsync(action, fileName, streamOptions, token);
+
+                return new EgressResult("uri", blobUri);
+            }
+
+            public override async Task<EgressResult> EgressAsync(
+                Func<Stream, CancellationToken, Task> action,
+                string fileName,
+                string contentType,
+                IEndpointInfo source,
+                CancellationToken token)
+            {
+                // TODO: Add metadata based on source
+                var streamOptions = new AzureBlobEgressStreamOptions();
+                streamOptions.ContentType = contentType;
+
+                var provider = new AzureBlobEgressProvider(_options, _loggerFactory.CreateLogger<AzureBlobEgressProvider>());
```

Comment:
>AzureBlobEgressProvider [](start = 35, length = 23)

Another feature from our discussion:
- Add a correlation id that can span Trigger responses and egresses.